blog/2022-11-09-nginx-monitoring.md

```diff
@@ -0,0 +1,101 @@
+
+- **Active Connections** - Which shows the currently active client connections and is an indication of the resource utilisation.
+!["Active Connections"](https://user-images.githubusercontent.com/96257330/200822049-6aebe009-78bc-48c8-aa2a-83cd90174004.png)
+
+- **Connection Status** - Which provides insights on what these connections are doing (**idle, reading, writing**)
+!["Connection Status](https://user-images.githubusercontent.com/96257330/200822841-60d29358-3e08-4221-a283-6b6a07a1829f.png)
+
+You can additionally group these metrics by instance to see how many connections are being handled per instance and in which state they are.
+![Connection Status per Instance](https://user-images.githubusercontent.com/96257330/200823083-2a35da9f-f6fc-4199-8c64-78b9417d4702.png)
+
+
+- **Accepted vs Handled Connections** - Which indicates if any connections are being dropped by calculating (**Accepted - Handled**) 
+![Accepted vs Handled](https://user-images.githubusercontent.com/96257330/200823600-b4a0c2d3-c430-4f94-b6e8-8ac1b4f2252b.png)
+
+You can also setup a custom alert to report any dropped connections on your NGINX.
+For example, the alert shown below will raise a Warning alert if there are 10-20% dropped connections and a Critical alert when the dropped connections go beyond 20%.
+
+```yaml
+  template: NGINX_Dropped_Connections_Exceeded
+        on: nginx.connections_accepted_handled
+     class: Utilization
+      type: NGINX
+ component: NGINX
+      calc: (($accepted - $handled) * 100) / ($accepted)
+     every: 1m
+     units: %
+      warn: $this > (($status >= $WARNING)  ? (10) : (20))
+      crit: $this > (($status == $CRITICAL) ? (20) : (30))
+     delay: down 15m multiplier 1.5 max 1h
+      info: The NGINX web server has exceeded the limit of dropped connections
+```
+## NGINX Error Metrics
+
+NGINX error metrics tell you how often your servers are returning errors instead of producing useful work. Client errors are represented by 4xx status codes, server errors with 5xx status codes.
+Although open source NGINX does not make error rates immediately available for monitoring, you can configure NGINX’s log module to write response codes in access log. More details on this are available on a related blog [How to monitor web servers and their performance?](https://blog.netdata.cloud/web-servers-and-their-performance/)
+
+## NGINX Performance Metrics
+
+The request time metric logged by NGINX records the processing time for each request, from the reading of the first client bytes to fulfilling the request. Long response times can point to problems upstream.
+Although open source NGINX does not make performance metrics immediately available for monitoring, you can configure NGINX’s log module to write request processing times in the access log. More details on this are available on a related blog [How to monitor web servers and their performance?](https://blog.netdata.cloud/web-servers-and-their-performance/)
```

Comment:
The last sentence here was already mentioned in the above section. Maybe avoid duplication and reference it once.

Reply:
Will do